### PR TITLE
Adding integration for retrofit integration

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
giphy

This PR adds documentation for the Retrofit integration

### CODE REVIEW

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            })
+```
+
+### Using only extensions functions
+
+It is possible to use extension functions for Retrofit Call so the code for the definition of the endpoints doesn't have to change.
+
+```kotlin
+    val call : Call<Response<String>>
+    call.runAsync(IO.async()) // Kind<F, Response<String>>
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: `Kind<ForIO, Response<String>>`

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          .fix() //IO<Response<String>>
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: //+ +IO

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          .unwrapBody(Either.applicativeError()) //Either<Throwable, ResponseMock>
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: `//+ +Either`

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+          .fix()
+          .fold({ throwable ->
+            //Ops!
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: //+ +O+o+ps!

Comment (leandroBorgesFerreira) [reply modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: What does the `+` means?

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+title: Retrofit
+permalink: /docs/integrations/retrofit/
+---
+
+## Retrofit
+
+{:.advanced}
+advanced
+
+Arrow contains a integration module for Retrofit so you can have a Arrow integration in your endpoints definition.
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: so you can use any synchronous or asynchronous datatype of your choice, like `Try`, `ObservableK`, `IO` or `DeferredK`.

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+Define your endpoints and define for CallK as the return Type:
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: `CallK`

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  @GET("testCallKResponse")
+  fun testCallKResponse(): CallK<ResponseMock>
+
+  @POST("testResponsePOST")
+  fun testIOResponsePost(): CallK<Unit>
+
+}
+```
+
+You can use CallK to have Async/MonadDefer/MonadError intances as your data wrapper.
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: `Async`, `MonadDefer`, `MonadError`, and make them link to their respective sites. You need to add a relative link, like the ones at the bottom.

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            .fold({ throwable ->
+              //Ops!
+            }, {
+              //Handle information
+            })
+```
+
+### Using only extensions functions
+
+It is possible to use extension functions for Retrofit Call so the code for the definition of the endpoints doesn't have to change.
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: Retrofit's `Call`

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+### Using CallK with ObservableK
+createApiClientTest(baseUrl)
+          .testCallK() //CallK
+          .async(ObservableK.async()) //Kind<F, Response<ResponseMock>>
+          .fix() //ObservableK<Response<ResponseMock>>
+```
+
+### Handling `Response` with Arrow
+
+Arrow provides a extension function for Response to handle it with Typeclasses. With `unwrapBody` you can extract the body to any kind of ApplicativeError.
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: for `Response<A>` to handle

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 10]
```diff
@@ -28,27 +28,27 @@ interface ApiClientTest {
...
 
-You can use CallK to have Async/MonadDefer/MonadError intances as your data wrapper.
+You can use `CallK` to have [`Async`]({{ '/docs/effects/async' | relative_url }}), [`MonadDefer`]({{ '/docs/effects/monaddefer' | relative_url }}) and [`MonadError`]({{ '/docs/effects/monaderror' | relative_url }}) intances as your data wrapper.
 
 ```kotlin
 ### Using CallK with IO
 createApiClientTest(baseUrl)
         .testCallK() //CallK
-        .async(IO.async()) //Kind<F, Response<ResponseMock>>
+        .async(IO.async()) //Kind<ForIO, Response<ResponseMock>>
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: One white space after the //, please

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 11]
```diff
@@ -28,27 +28,27 @@ interface ApiClientTest {
...
           .testCallK() //CallK
-          .async(ObservableK.async()) //Kind<F, Response<ResponseMock>>
+          .async(ObservableK.async()) //Kind<ForObservableK, Response<ResponseMock>>
           .fix() //ObservableK<Response<ResponseMock>>
 ```
 
 ### Handling `Response` with Arrow
 
-Arrow provides a extension function for Response to handle it with Typeclasses. With `unwrapBody` you can extract the body to any kind of ApplicativeError.
+Arrow provides a extension function for Response<A> to handle it with Typeclasses. With `unwrapBody` you can extract the body to any kind of ApplicativeError.
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: Backticks to highlight `Response`

File: modules/docs/arrow-docs/docs/docs/integrations/rx2/README.md [hunk 12]
```diff
@@ -85,7 +85,7 @@ subject.value()
 
 ### Observable comprehensions
 
-The library provides instances of [`MonadError`]({{ '/docs/typeclasses/monaderror' | relative_url }}) and [`MonadDefer`]({{ '/docs/effects/monaddefer' | relative_url }}).
+The library provides instances of [MonadError]({{ '/docs/typeclasses/monaderror' | relative_url }}) and [`MonadDefer`]({{ '/docs/effects/monaddefer' | relative_url }}).
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/rx2/README.md]: Backticks for MonadError.

File: modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md [hunk 13]
```diff
@@ -9,9 +9,9 @@ permalink: /docs/integrations/retrofit/
 {:.advanced}
 advanced
 
-Arrow contains a integration module for Retrofit so you can have a Arrow integration in your endpoints definition.
+Arrow contains a integration module for Retrofit so you can use any synchronous or asynchronous datatype of your choice, like Try, ObservableK, IO or DeferredK.
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/integrations/retrofit/README.md]: Backticks and links for Try, ObservableK, IO, and DeferredK.

File: modules/docs/arrow-docs/docs/docs/integrations/rx2/README.md (hunk 12 above)
Comment (leandroBorgesFerreira) [reply modules/docs/arrow-docs/docs/docs/integrations/rx2/README.md]: rx2??

Comment (pakoito) [reply modules/docs/arrow-docs/docs/docs/integrations/rx2/README.md]: These -> `

Comment (leandroBorgesFerreira) [reply modules/docs/arrow-docs/docs/docs/integrations/rx2/README.md]: Oh I see. I shouldn't have edited the file in the rx2. It was a mistake

**Comment by pakoito** [comment]
Merge as soon as CI passes!

**Pull request merged**